### DISCUSSION TITLE
How do I change the server/api path?

### DISCUSSION THREAD

**Discussion by kdydesign** [nuxt/nuxt#19813] [Questions]
Hi.

```
nuxt 3.3.1
```

I want to configure the process of processing restapi at the backend in 'server/api'.

We proxy for '/api' when calling restapi internally. Therefore, 'server/api' fixed in the server directory cannot be used. (Do not call server/api using devProxy in nitro)

``` javascript
export default defineNuxtConfig({
    ssr: false,
    nitro: {
        devProxy: {
            '/api/': {
                target: 'http://x.x.x.x/api/'
            }
        }
    }
})
```

``` javascript
// Do not call server/api
await useFetch('/api/user/login', { params })
```

**~/server/api/user/login.[...].js**
``` javascript
export default defineEventHandler(async (event) => {
    // I want to call a proxy here.
    return await $fetch('/api/call/au/user', {...})
})
```

My question is, can I change the name of 'api' in the server directory or create another folder to auto-import?

**Comment by danielroe** [comment]
You can add paths not prefixed with `/api` in the `~/server/routes` folder. They will be auto registered without a prefix. So `~/server/routes/myapi.ts` will be accessible at `/myapi`.

**Reply by kdydesign** [reply]
Thank you.

Let me ask you one more question.

I did not understand the role of `api` and `routes` and `middleware` in the server directory. Why do you need these? Is it simply to distinguish between the logical part of the server and the client?

**Reply by kdydesign** [reply]
I tested it by changing it to server/route. However, the url requesting the server is calling the url used in `useFetch` regardless of my intention.

I want to make a request to the server through `$fetch` inside `defineEventHandler`.

``` html
// login.vue
<script setup>
 const { data} = await useFetch('/user/login', {
     method: 'POST',
     body: {}
 })
</script>
```

``` javascript
// server/routes/user/login.post.js
export default defineEventHandler(aysnc (event) => {
    // I want to call server api here.
    const res = await $fetch('/api/user/login', {
        // ....
    })
})
```

``` javascript
// nuxt.config.js
export default defineNuxtConfig({
    ssr: false,
    devProxy: {
        '/api/': {
            target: 'http://x.x.x.x/api/',
            changeOrigin: true
        }
    }
})

```

I called `/api/user/login` to the server, but I'm actually calling `user/login` when I see devtools. What's going on here?

```
// network
Request URL: http://x.x.x.x/user/login
```
